mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -2022,6 +2025,100 @@ func TestClientSideEncryptionProse(t *testing.T) {
 		})
 	})
 
+	mt.RunOpts("18. Azure IMDS Credentials", noClientOpts, func(mt *mtest.T) {
+		buf := make([]byte, 0, 256)
+		kmsProvidersMap := map[string]map[string]interface{}{
+			"azure": {},
+		}
+		p, err := bson.MarshalAppendWithRegistry(bson.DefaultRegistry, buf[:0], kmsProvidersMap)
+		assert.Nil(mt, err, "error in MarshalAppendWithRegistry: %v", err)
```

Comment:
Optional: If an error here would make the rest of the test invalid, consider using `require.NoError` to record the failure and stop the test instead of continuing.

E.g.
```go
require.NoError(mt, err, "MarshalAppend error")
```

That suggestion applies for all "nil error" assertions that use `assert`.